pkg/terminal/command.go

```diff
@@ -1796,11 +1825,23 @@ func formatBreakpointAttrs(prefix string, bp *api.Breakpoint, includeTrace bool)
 }
 
 func setBreakpoint(t *Term, ctx callContext, tracepoint bool, argstr string) ([]*api.Breakpoint, error) {
+	var (
+		cond string
+		spec string
+
+		requestedBp = &api.Breakpoint{}
+	)
+
+	r := regexp.MustCompile(`^if | if `)
```

Comment:
We accept arbitrary expressions here so this would fail for something like `break *m[" if "]`. I think the only way to do this is to do the same thing we do for `call` and assignments: try to parse the expression as a normal location expression, and if there is a syntax error look up where the error is in the string if it is an `if`. See pkg/proc/evalop/evalcompile.go:59.

Reply:
PS. another situation where this fails is `break /a/path/that/happens/to/have an if inside/it.go:20`.